src/Sema.zig

```diff
@@ -19767,6 +19809,14 @@ fn analyzeSlice(
                 slice_ty = ptr_ptr_child_ty;
                 array_ty = ptr_ptr_child_ty;
                 elem_ty = ptr_ptr_child_ty.childType();
+
+                if (ptr_ptr_child_ty.ptrSize() == .C) {
+                    if (try sema.resolveMaybeUndefVal(block, ptr_src, ptr_or_slice)) |ptr_val| {
+                        if (!ptr_val.isUndef() and ptr_val.isNull()) {
```

Comment:
This is a case for `resolveDefinedValue` instead of `resolveMaybeUndefVal` because according the proposed undef semantics, if there exists an adversarial operand value that would cause Illegal Behavior, and that operand is `undefined`, then Illegal Behavior is invoked. In other words, `isUndef()` is treated equivalent to `isNull()` in this case.